src/DotLiquid/Util/StringExtensionMethods.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace DotLiquid.Util
+{
+    public static class StringExtensionMethods
+    {
+        private static readonly Regex IntegerRegex = R.C(R.Q(@"^([+-]?\d+)$"));
+        private static readonly Regex NumericRegex = R.C(R.Q(@"^([+-]?\d[\d\.|\,]+)$"));
+
+        public static bool TryParseToNumericType(this string value, IFormatProvider formatProvider, out object convertedValue)

```

Comment:
I like the idea, but maybe we should have INumeric and expose Math.Floor etc. on the INumeric interface. This would avoid need of type-casting.

Reply:
Sounds like another feature request! :)